source/ByteBuf.cpp

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <aws/crt/ByteBuf.h>
+
+namespace Aws
+{
+    namespace Crt
+    {
+        ///////////////////////////////////////////////////////////////////////////////////
+
+        ByteCursor::ByteCursor() noexcept : m_cursorPtr(&m_cursor) { AWS_ZERO_STRUCT(m_cursor); }
```

Comment:
trivial: In c++ you can zero out a struct in the member-initializer-list like so
`: m_cursor{0}`

Reply:
nitpick: `m_cursor{0}` initializes the **first** member to 0 and default initialize the rest of the members. However, in this case they're the same.
But `m_cursor{42}` would _not_ initialize all the members to 42!!!
Instead, it initializes the first member to 42 and default initialize the rest of the members.

So, `m_cursor{}` is just as good to default initialize the struct. effectively zeroing it out.